test/units/module_utils/facts/system/test_pkg_mgr.py

```diff
@@ -0,0 +1,56 @@
+from ansible.module_utils.facts.system.pkg_mgr import PkgMgrFactCollector
+
+
+_FEDORA_FACTS = {
+    "ansible_distribution": "Fedora",
+    "ansible_distribution_major_version": 38,  # any version where yum isn't default
+    "ansible_os_family": "RedHat"
+}
+
+# NOTE pkg_mgr == "dnf" means the dnf module for the dnf version 4, which is accessible via /usr/bin/dnf-3
+
+def test_default_dnf_version_detection_fedora_dnf4(mocker):
+    mocker.patch("os.path.exists", lambda p: p in ("/usr/bin/dnf", "/usr/bin/dnf-3"))
+    mocker.patch("os.path.realpath", lambda p: {"/usr/bin/dnf": "/usr/bin/dnf-3"}.get(p, p))
+    assert PkgMgrFactCollector().collect(collected_facts=_FEDORA_FACTS).get("pkg_mgr") == "dnf"
+
+
+def test_default_dnf_version_detection_fedora_dnf5(mocker):
+    mocker.patch("os.path.exists", lambda p: p in ("/usr/bin/dnf", "/usr/bin/dnf5"))
+    mocker.patch("os.path.realpath", lambda p: {"/usr/bin/dnf": "/usr/bin/dnf5"}.get(p, p))
+    assert PkgMgrFactCollector().collect(collected_facts=_FEDORA_FACTS).get("pkg_mgr") == "dnf5"
+
+
+def test_default_dnf_version_detection_fedora_dnf4_both_installed(mocker):
+    mocker.patch("os.path.exists", lambda p: p in ("/usr/bin/dnf", "/usr/bin/dnf-3", "/usr/bin/dnf5"))
+    mocker.patch("os.path.realpath", lambda p: {"/usr/bin/dnf": "/usr/bin/dnf-3"}.get(p, p))
+    assert PkgMgrFactCollector().collect(collected_facts=_FEDORA_FACTS).get("pkg_mgr") == "dnf"
+
+
+def test_default_dnf_version_detection_fedora_dnf4_microdnf5_installed(mocker):
+    mocker.patch(
+        "os.path.exists",
+        lambda p: p in ("/usr/bin/dnf", "/usr/bin/microdnf", "/usr/bin/dnf-3", "/usr/bin/dnf5")
+    )
+    mocker.patch(
+        "os.path.realpath",
+        lambda p: {"/usr/bin/dnf": "/usr/bin/dnf-3", "/usr/bin/microdnf": "/usr/bin/dnf5"}.get(p, p)
+    )
+    assert PkgMgrFactCollector().collect(collected_facts=_FEDORA_FACTS).get("pkg_mgr") == "dnf"
+
+
+def test_default_dnf_version_detection_fedora_dnf4_microdnf(mocker):
+    mocker.patch("os.path.exists", lambda p: p in ("/usr/bin/microdnf", "/usr/bin/dnf-3"))
+    mocker.patch("os.path.realpath", lambda p: {"/usr/bin/microdnf": "/usr/bin/dnf-3"}.get(p, p))
```

Comment:
Note: This should not happen on a real system. microdnf was never a symlink to dnf-3. I'm not sure why a user would manually create one.